src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -93,9 +93,7 @@ module SimulationCode =
     let isGeneric context (n: QsQualifiedName) =
         if context.allCallables.ContainsKey n then
             let signature = context.allCallables.[n].Signature
-            let tIn = signature.ArgumentType
-            let tOut = signature.ReturnType
-            hasTypeParameters [tIn;tOut]
+            not signature.TypeParameters.IsEmpty
```

Comment:
I suppose `signature.TypeParameters` includes both, `tIn` and `tOut`?

Reply:
The `signature.TypeParameters` is different from the ArgumentType or the ReturnType. Instead of counting the number of unique type parameter references in the ArgumentType and ReturnType of the callable to determine if it is generic, the new way I am presenting is to count the number of type parameters listed as part of the signature, which seems more straightforward to me as a means of checking if something is generic. This also avoids the underlying problem of the bug, which was when type parameters of the callable are not found in the argument type or return type, they would not be counted and the generated class would not be correct, causing a build-time error for the Q# code.